src/text-editor.coffee

```diff
@@ -2587,6 +2587,18 @@ class TextEditor extends Model
       maintainClipboard = true
     return
 
+
+  # Essential: For each selection, only copy hightlighted text. Line is coptied if no selected text and cursor is at the beginning of the line
+  copyOnlySelectedText: ->
+    maintainClipboard = false
+    for selection in @getSelectionsOrderedByBufferPosition()
+      if not selection.isEmpty()
+        selection.copy(maintainClipboard, true)
+      else if selection.isEmpty() and selection.cursor.isAtBeginningOfLine()
+        @copySelectedText()
```

Comment:
Just wondering why you decided to call `copySelectedText()` here instead of inlining the part for copying the line? Calling `copySelectedText()` will cause _all selections_ to be processed again within that function, not just the selection currently being processed within `copyOnlySelectedText()`, right? While that normally won't be a performance problem, I think avoiding doing the same thing over and over again (since this would be a loop nested within the same loop) is always nice. 

I also wonder if calling `copySelectedText()` here might cause problems in some cases because _all selections_ will be processed according to the rules of that function, and not just by the rules of `copyOnlySelectedText()`? I'm thinking of cases with multiple selections, where one selection is empty and at the beginning of the line, and the other is empty but not at the beginning of the line.